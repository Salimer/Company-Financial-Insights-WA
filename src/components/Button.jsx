import PropTypes from 'prop-types';

const Button = ({ children }) => (
  <button className=" bg-buttonFirstColor p-5 text-white" type="button">{children}</button>
);
export default Button;

Button.propTypes = {
  children: PropTypes.string.isRequired,
};
