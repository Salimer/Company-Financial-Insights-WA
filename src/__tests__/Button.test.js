import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Button from '../components/Button';

describe('Button component', () => {
  it('renders the button correctly', () => {
    const props = {
      children: 'Test Button',
      url: 'test-url',
      iconName: 'test-icon',
    };
    const { container } = render(
      <MemoryRouter>
        <Button {...props} />
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
