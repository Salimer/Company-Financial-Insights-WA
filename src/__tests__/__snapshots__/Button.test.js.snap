// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders the button correctly 1`] = `
<a
  class="bg-buttonFirstColor p-5 text-white flex flex-col justify-center items-center border-white border"
  href="/test-url"
>
  <span>
    <ion-icon
      name="test-icon"
      size="large"
    />
  </span>
  <span
    class=" text-lg tracking-wider font-bold"
  >
    Test Button
  </span>
</a>
`;
